reader_test.go

```diff
@@ -5,20 +5,39 @@ import (
 	"testing"
 )
 
-// TestFedWireMessage validates reading an FedWireMessage file
-func TestFedWireMessageFileRead(t *testing.T) {
-	f, err := os.Open("./test/testdata/fedWireMessage.txt")
-	if err != nil {
-		t.Errorf("%T: %s", err, err)
-	}
-	defer f.Close()
-	r := NewReader(f)
-	_, err = r.Read()
+// TestRead reads wire Files with different BusinessFunctionCodes
+func TestRead(t *testing.T) {
+	t.Run("BankTransfer", testRead("./test/testdata/fedWireMessage-BankTransfer.txt"))
+	t.Run("CustomerTransfer", testRead("./test/testdata/fedWireMessage-CustomerTransfer.txt"))
+	t.Run("CustomerTransferPlus", testRead("./test/testdata/fedWireMessage-CustomerTransferPlus.txt"))
+	t.Run("CheckSameDaySettlement", testRead("./test/testdata/fedWireMessage-CheckSameDaySettlement.txt"))
+	t.Run("DepositSendersAccount", testRead("./test/testdata/fedWireMessage-DepositSendersAccount.txt"))
+	t.Run("FEDFundsReturned", testRead("./test/testdata/fedWireMessage-FEDFundsReturned.txt"))
+	t.Run("FEDFundsSold", testRead("./test/testdata/fedWireMessage-FEDFundsSold.txt"))
+	t.Run("DrawdownRequest", testRead("./test/testdata/fedWireMessage-DrawdownRequest.txt"))
+	t.Run("BankDrawdownRequest", testRead("./test/testdata/fedWireMessage-BankDrawdownRequest.txt"))
+	t.Run("CustomerCorporateDrawdownRequest", testRead("./test/testdata/fedWireMessage-CustomerCorporateDrawdownRequest.txt"))
+	t.Run("ServiceMessage", testRead("./test/testdata/fedWireMessage-ServiceMessage.txt"))
```

Comment:
I'm surprised this passes on Windows. I use `path/filepath.Join` to handle `/` vs `\`.

Reply:
I know I saw you had done that.  This works though.